src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -44,10 +44,12 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     double lat = convertToDouble(latString);
     double lng = convertToDouble(lngString);
     MapLocation userLocation = new MapLocation(lat, lng);
-    smallCityService.createUserServiceWithGeolocation(userLocation);
+    String product = request.getParameter("product");
+    smallCityService.createUserServiceWithGeolocation(userLocation, product);
     } catch(NullPointerException e) {
```

Comment:
nit: Hate to be that guy but need a space after catch

Reply:
Fixed